src/Simulation/QsharpCore/Diagnostics/Facts.qs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation.
```

Comment:
I am confused why this file is part of this PR.

Reply:
Ah, sorry for not clarifying. I needed some basic facts for unit testing, such that this PR moves `Fact` and `Contradiction` from quantumlibraries to qsharpcore. This has no user-facing impact, since the Microsoft.Quantum.Standard project is brought in by default, but helps with dependency ordering.